tests/components/zha/test_light.py

```diff
@@ -63,6 +67,43 @@
 }
 
 
+@asynctest.mock.patch(
+    "homeassistant.components.zha.light.Light._refresh", new=MagicMock()
```

Comment:
We shouldn't patch the entity. Why do we need to do that? Try to patch in the periphery of possible, eg the device library and assert that it has been called.

Reply:
It’s an entity function that’s called by a custom polling method. The methods it calls also get called elsewhere. How would I go about that in a deterministic way? The point of the test is to test the custom polling in the entity class itself so in this case why isn’t this appropriate?

Reply:
No, the point of the test is to test that the entity gets new state info from the device at the custom interval. The point is not to test the details of how that is done inside the entity. By patching in the periphery we can assert the correct behavior while keeping the test robust for changes in the implementation details.

Reply:
so, again, just trying to understand here. I want to make sure that said new state comes from `refresh` and no other way... so I shouldn’t ensure that `refresh`  is actually called when it should be. I should go further down and assert that something else was called later? To me that seems odd. in normal cases I completely understand asserting against the state machine. In this case I don’t even care what the state is. All I care about is that the code that was written to call `refresh` did what it was supposed to. Am I off in left field?

Reply:
We shouldn't care about that the state comes from the refresh method. We should just care about that the entity gets new state info from the device at the correct interval. How exactly that is done, isn't important. It should just happen.

Reply:
I updated the test to do what you asked but I respectfully disagree with you on the above points. I think your POV is that you think I am testing `async_track_time_interval` itself and not what I expect it to do? Whereas my POV is that the only code I really want to test with this _unit_ test is that the code I specifically added to `async_added_to_hass` that actually attached the interval handler and that it is calling the expected method at the intended interval specific to my class. (That's all that changed here... the state from the polls is completely irrelevant IMO so long as we actually poll when expected) This is a unit test and everything I have learned up to this point says they should be kept as small as possible in scope. What you are saying is to essentially test the side effects of what was changed? There are reams of code in between what I just added and what actually hits the state machine or the backend libs. I could make completely unrelated changes in the future that negate this test unknowingly by causing the asserted calls or state to match what this is looking for (it's all async some other events could sneak in that have unintended effects) and then a separate subsequent change could screw up the code in `async_added_to_hass` in the light class and we would never know because by doing it this way we aren't testing just that code... We are testing all of the machinery in-between what was actually changed and all of the things necessary to take the data from the state poll and correctly use it in HA.

Reply:
aside from my diatribe above, I have updated the test to assert against the zigpy calls. Is this better?